device.go

```diff
@@ -32,6 +32,7 @@ type device struct {
 	BootEnvReadWriter
 	Commander
 	*partitions
+	hasUpdateCb func(d *device) (bool, error)
```

Comment:
@bboozzoo right, I haven't thought about that. Makes perfect sense. @oleorhagen I think what @bboozzoo is suggesting makes lot of sense.